microsoft/testsuites/vm_extensions/cvt.py

```diff
@@ -0,0 +1,284 @@
+    )
+
+    for file in file_list:
+        log.info(f"Copying file {file} to {log_path}")
+        try:
+            file_name = file.split("/")[-1]
+            node.shell.copy_back(
+                node.get_pure_path(file),
+                log_path / f"{file_name}",
+            )
+        except FileNotFoundError:
+            log.error(f"File {file} doesn't exist.")
+
+
+def _run_cvt_tests(
+    log: Logger,
+    node: Node,
+    log_path: Path,
+    variables: Dict[str, Any],
+    os: str,
+    cvt_script: CustomScriptBuilder,
+) -> Optional[int]:
+    cvt_bin = "indskflt_ct"
+    container_sas_uri = variables.get("cvtbinaries_sasuri", "")
+    if not container_sas_uri:
+        log.error("sas uri for cvt binary is empty.")
+        raise SkippedException("cvt binary is not provided.")
+    cvt_binary_sas_uri = container_sas_uri.replace(
+        "?", "/cvtbinaries/indskflt_ct_" + os + "?"
+    )
+
+    cvt_root_dir = str(node.working_path) + "/LisaTest/"
+    cvt_download_dir = cvt_root_dir + "cvt_files/"
+
+    mkdir = node.tools[Mkdir]
+    wget = node.tools[Wget]
+
+    mkdir.create_directory(cvt_download_dir, sudo=True)
+    download_path = wget.get(
+        url=f"{cvt_binary_sas_uri}",
+        filename=cvt_bin,
+        file_path=cvt_download_dir,
+        sudo=True,
+    )
+
+    cvt_md5sum = node.execute(f"md5sum {download_path}", shell=True, sudo=True)
+    log.info(f"md5sum '{download_path}' : '{cvt_md5sum}'")
+
+    result = _run_script(
+        node=node, log=log, test_dir=cvt_download_dir, cvt_script=cvt_script
+    )
+    _copy_cvt_logs(
+        node=node,
+        log=log,
+        test_dir=Path(node.working_path),
+        log_path=log_path,
+    )
+    return result.exit_code
+
+
+@TestSuiteMetadata(
+    area="cvt",
+    category="functional",
+    description="""
+    This test is used to validate the functionality of ASR driver.
+    """,
+    requirement=simple_requirement(unsupported_os=[]),
+)
+class CVTTest(TestSuite):
+    TIMEOUT = 21600
+
+    @TestCaseMetadata(
+        description="""
+        this test validate the functionality of ASR driver by verifying
+        integrity of a source disk with respect to a target disk
+        """,
+        priority=1,
+        timeout=TIMEOUT,
+    )
+    def verify_asr_by_cvt(
+        self,
+        node: Node,
+        log: Logger,
+        log_path: Path,
+        variables: Dict[str, Any],
+    ) -> None:
+        _init_disk(node=node, log=log)
+        os = _get_os_info_from_extension(node=node, log=log)
+        _install_asr_extension_distro(node=node, log=log, os=os)
+        result = _run_cvt_tests(
+            node=node,
+            log=log,
+            log_path=log_path,
+            variables=variables,
+            os=os,
+            cvt_script=self._cvt_script,
+        )
+        log.info(f"ASR CVT test completed with exit code '{result}'")
+        assert_that(result).described_as("ASR CVT test failed").is_equal_to(0)
```

Comment:
If the output of script is printed together, it's easier to troubleshoot in future. The failure message can be provided by dashboard, so it doesn't need to look into log files. But make it shorter than 200 chars.

Reply:
@mohshei786 It looks like the error message is not improved. Any detailed error can be printed with this? @LiliDeng